pkg/drivers/hyperkit/driver.go

```diff
@@ -240,6 +245,48 @@ func (d *Driver) Start() error {
 	return nil
 }
 
+func (d *Driver) checkForPidFile() error {
+	stateDir := filepath.Join(d.StorePath, "machines", d.MachineName)
+	pidFile := filepath.Join(stateDir, pidFileName)
+
+	if _, err := os.Stat(pidFile); err != nil {
+		log.Infof("clean start, hyperkit pid file doesn't exist: %s", pidFile)
+		return nil
+	}
+
+	log.Warnf("hyperkit pid file exists: %s", pidFile)
+	// we have a pid file, hyperkit might be running!
+	// Make sure the pid written by hyperkit is the same as in the json
+	content, err := ioutil.ReadFile(pidFile)
+	if err != nil {
+		return errors.Wrapf(err, "reading pidfile %s", pidFile)
+	}
+	pid, err := strconv.Atoi(string(content[:]))
+	if err != nil {
+		return errors.Wrapf(err, "parsing pidfile %s", pidFile)
+	}
+
+	p, _ := os.FindProcess(pid) //noop on linux
+	// Sending a signal of 0 can be used to check the existence of a process.
+	if err := p.Signal(syscall.Signal(0)); err == nil {
+		return fmt.Errorf("something is not right...please stop all minikube instances, seemingly a hyperkit server is already running with pid %d", pid)
```

Comment:
Does it matter at all if this pid is hyperkit, and not a /bin/ls process that has reused the pid number?

If not, you might need to use a library like https://github.com/mitchellh/go-ps check that the process name contains hyperkit:

```
p := ps.FindProcess(pid)
if !strings.Contains(p.Executable(), "hyperkit") {
  .. stale pid file ...
}
```

Reply:
It does matter - as if there is a hyperkit instance with this same exact PID - as there is a non-zero chance that we are looking at some kind of race condition with another starting minikube instance with the same name and then we should abort. There is also a chance it's a different minikube instance with different name - but then again - I can't really differentiate between the two so I'd rather fail here. 

If the PID is reused by another non-hyperkit instance, then I don't care, I can actually remove the pid file and have hyperkit restart with another PID.